CodeEdit/Documents/WorkspaceCodeFileView.swift

```diff
@@ -30,26 +31,65 @@ struct WorkspaceCodeFileView: View {
                 }
                 return file.tabID == workspace.selectionState.selectedId
             }) {
-                if let codeFile = workspace.selectionState.openedCodeFiles[item] {
-                    CodeFileView(codeFile: codeFile)
-                        .safeAreaInset(edge: .top, spacing: 0) {
-                            VStack(spacing: 0) {
-                                BreadcrumbsView(file: item, tappedOpenFile: workspace.openTab(item:))
-                                Divider()
-                            }
-                        }
-                } else {
-                    Text("No Editor")
-                        .font(.system(size: 17))
-                        .foregroundColor(.secondary)
-                        .frame(minHeight: 0)
-                        .clipped()
+                if let fileItem = workspace.selectionState.openedCodeFiles[item] {
+                    switch fileItem.typeOfFile {
+                    case UTType.image:
+                        imageFileView(fileItem, for: item)
+                    case UTType.text:
+                        codeFileView(fileItem, for: item)
+                    default:
+                        otherFileView(fileItem, for: item)
+                    }
                 }
+            } else {
+                Text("No Editor")
+                    .font(.system(size: 17))
+                    .foregroundColor(.secondary)
+                    .frame(minHeight: 0)
+                    .clipped()
             }
         }
         .frame(maxWidth: .infinity, maxHeight: .infinity)
     }
 
+    @ViewBuilder
+    private func codeFileView(
+        _ codeFile: CodeFileDocument,
+        for item: WorkspaceClient.FileItem
+    ) -> some View {
+        CodeFileView(codeFile: codeFile)
+            .safeAreaInset(edge: .top, spacing: 0) {
+                VStack(spacing: 0) {
+                    BreadcrumbsView(file: item, tappedOpenFile: workspace.openTab(item:))
+                    Divider()
+                }
+            }
```

Comment:
This is because if I use safeAreaInset() here, images in ScrollView will have a 1 pixel space on the top.
<img width="739" alt="Screen Shot 2022-07-17 at 14 09 31" src="https://user-images.githubusercontent.com/44807628/179383595-9308f018-9952-4898-8b3f-67cc093d558d.png">

I think using VStack is making more sense.

Reply:
I fully agree with you, and I think we should use only `VStack`. Using `safeAreaInset` seems to be unneeded to me.

Reply:
@shibotong @Matejkob We need to use `safeAreaInset` so that we will see the blurred effect in the toolbar when there is overflow. The above screenshot reveals a defect in the tab bar that needs its own issue and needs to be fixed.

Reply:
Thank you @austincondiff I have changed these views to `safeAreaInset`. I think its ready to merge now